README.txt

Feel free to add some documentation or simply add a link to the online manual.

Added a line to the end of the doc.

test

ChangeLog
#
# ChangeLog for /tags/wec_sermons_0_9_3 
# 
# Generated by Trac 0.10.4
# 07/27/07 18:30:04
#


07/26/07 15:19:40 brian [386]
	* trunk/wec_sermons/ext_emconf.php (modified)
	- Updated version to 0.9.3 for release

07/21/07 21:32:01 brian [385]
	* trunk/wec_sermons/pi1/class.tx_wecsermons_pi1.php (modified)
	- Bugfix: Corrected a mis-spelling in $startDate variable when
	calculating start date for plugin filter. Also added the check for
	n/n/n or n-n-n date formats.

07/19/07 20:25:05 brian [384]
	* trunk/wec_sermons/doc/manual.sxw (modified)
	* trunk/wec_sermons/pi1/class.tx_wecsermons_pi1.php (modified)
	- Improvement: Added a check for proper loading of searchbox subpart
	- New Feature: Added processing to the startDate and endDate
	properties within listView and latestView, which accepts the function
	'date()' to calculate todays date. This allows one to specify a
	moving window of records to be shown, such as 'date()-365'.
	- Bugfix: startDate and endDate now filter series records, if a non-
	grouped list of series records is chosen.
	- Updated documentation
	- Removed trailing spaces

07/19/07 02:26:28 brian [383]
	* trunk/wec_sermons/pi1/class.tx_wecsermons_pi1.php (modified)
	- Rolled back bugfix for series season and series topic logic checks,
	as this would cause mysql errors.

07/18/07 23:48:03 brian [382]
	* trunk/wec_sermons/doc/manual.sxw (modified)
	* trunk/wec_sermons/pi1/class.tx_wecsermons_pi1.php (modified)
	- Udated documentation with proper occurrence_date marker name
	- Bugfix: getSeriesResources and getSermonResources functions were
	filtering out single view records by the pid. The starting points
	should be obeyed for views other than single views.
	- Bugfix: Corrected series_season and series_topics subpart marker
	processing, as they were not saving current and previous row before
	changing context to related tables.

07/14/07 20:43:21 brian [379]
	* trunk/wec_sermons/doc/manual.sxw (modified)
	- Updated change log in manual.sxw

07/09/07 15:28:31 brian [364]
	* trunk/wec_sermons/doc/manual.sxw (modified)
	* trunk/wec_sermons/pi1/class.tx_wecsermons_pi1.php (modified)
	- New Feature: Modified the results browser so that it will render
	only if there is more than one page of results to be shown.

07/09/07 15:09:35 brian [363]
	* trunk/wec_sermons/doc/manual.sxw (modified)
	* trunk/wec_sermons/pi1/class.tx_wecsermons_pi1.php (modified)
	* trunk/wec_sermons/static/constants.txt (modified)
	* trunk/wec_sermons/static/setup.txt (modified)
	- New Feature: Added the Smart Display option, allowing a user to
	turn off the auto-detection of SINGLE view requests. This is needed
	when you have two plugins enabled on a page, one set to always show a
	list or latest view, and the other showing a single view.

07/03/07 22:14:28 brian [362]
	* trunk/wec_sermons/pi1/class.tx_wecsermons_pi1.php (modified)
	- BUGFIX: Added a check for ###CONTENT### subpart marker for single,
	latest, and list views. If this is not available then it could throw
	Apache into a loop.

07/03/07 14:28:29 brian [361]
	* trunk/wec_sermons/pi1/class.tx_wecsermons_pi1.php (modified)
	- BUGFIX: list and latest views were using piVars['recordType'] to
	specify the current table, when this should be done from plugin or
	typoscript. Configured to set and use $this-&gt;internal['currentTable']

	- BUGFIX: Processing the ITEM part of a template when it was not
	present in grouped views caused the plugin to fail. Created branch
	that if ITEM subpart is not present, we do not try to process.

07/02/07 20:35:05 brian [356]
	* trunk/wec_sermons/ext_tables.php (modified)
	- BUGFIX: Resources were not ordering themselves by crdate
	descending, rather by crdate ascending and title descending. Moved
	the DESC keyword to the proper place.

06/29/07 04:45:30 brian [351]
	* trunk/wec_sermons/pi1/class.tx_wecsermons_pi1.php (modified)
	- BUGFIX: Group Table setting from TypoScript was misnamed and
	looking at wrong scope in the configuration. Set to check
	plugin.tx_wecsermons_pi1.groupTable propery

06/29/07 03:58:00 brian [350]
	* trunk/wec_sermons/pi1/class.tx_wecsermons_pi1.php (modified)
	- BUGFIX: Layout code was not being set in the proper place for it to
	work through the TypoScript setting. Placed this in each view's
	function so that it is processed.

06/20/07 19:06:13 brian [345]
	* trunk/wec_sermons/doc/manual.sxw (modified)
	- Updated manual.sxw with update on &#34;Template Name&#34; field of SMS
	Resource Types, and how this field defaults to TEMPLATE_+ Marker Name

06/16/07 23:36:43 brian [343]
	* trunk/wec_sermons/pi1/class.tx_wecsermons_pi1.php (modified)
	- BUGFIX: Corrected an order of operations problem with grouped
	views. We were switching context to sermons before series were done.

	- BUGFIX: Corrected grouped output query. Searching for series uids
	within the sermons record was not returning the proper results. Now
	query uses find_in_set function to find series uids within the
	series.uid field.

06/12/07 16:00:26 brian [339]
	* trunk/wec_sermons/static/constants.txt (modified)
	- Constants document change

06/11/07 21:49:14 brian [338]
	* trunk/wec_sermons/pi1/class.tx_wecsermons_pi1.php (modified)
	* trunk/wec_sermons/static/constants.txt (modified)
	* trunk/wec_sermons/static/setup.txt (modified)
	- Added constant for xmlview maxdetailResults to limit the number of
	items in an xml feed.

06/11/07 18:10:15 brian [337]
	* trunk/wec_sermons/pi1/class.tx_wecsermons_pi1.php (modified)
	* trunk/wec_sermons/static/setup.txt (modified)
	- Added trim to title, description, and author of RSS feed to clean
	up any white space.
	- xmlView now merges resource fields with the sermon fields, so that
	all are available to TypoScript.

06/01/07 21:01:03 brian [332]
	* trunk/wec_sermons/ChangeLog (modified)
	* trunk/wec_sermons/doc/manual.sxw (modified)
	* trunk/wec_sermons/ext_emconf.php (modified)
	- Updated suggested and required constraints in ext_emconf.php

ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext: "wec_sermons"
#
# Auto generated 14-03-2007 16:09
#
# Manual updates:
# Only the data in the array - anything else is removed by next write.
# "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'WEC Sermon Management System',
	'description' => 'Provides centralized management of online resources associated with a sermon',
	'category' => 'plugin',
	'author' => 'Web-Empowered Church Team',
	'author_company' => 'Foundation For Evangelism',
	'author_email' => 'sermon@webempoweredchurch.org',
	'shy' => '',
	'dependencies' => 'wec_api',
	'conflicts' => '',
	'priority' => '',
	'module' => '',
	'state' => 'beta',
	'internal' => '',
	'uploadfolder' => 1,
	'createDirs' => '',
	'modify_tables' => '',
	'clearCacheOnLoad' => 1,
	'lockType' => '',
	'version' => '0.9.4rc1',
	'_md5_values_when_last_written' => 'a:144:{s:9:"ChangeLog";s:4:"4842";s:10:"README.txt";s:4:"9386";s:9:"Thumbs.db";s:4:"5af3";s:39:"class.tx_wecsermons_resourceTypeTca.php";s:4:"e553";s:31:"class.tx_wecsermons_xmlView.php";s:4:"10bb";s:12:"ext_icon.gif";s:4:"0e1f";s:17:"ext_localconf.php";s:4:"d9ae";s:14:"ext_tables.php";s:4:"d98a";s:14:"ext_tables.sql";s:4:"2c55";s:28:"ext_typoscript_constants.txt";s:4:"d41d";s:24:"ext_typoscript_setup.txt";s:4:"d41d";s:19:"flexform_ds_pi1.xml";s:4:"aa81";s:37:"icon_tx_wecsermons_resource_types.gif";s:4:"3a1a";s:32:"icon_tx_wecsermons_resources.gif";s:4:"86f1";s:30:"icon_tx_wecsermons_seasons.gif";s:4:"c430";s:29:"icon_tx_wecsermons_series.gif";s:4:"67c9";s:30:"icon_tx_wecsermons_sermons.gif";s:4:"d132";s:31:"icon_tx_wecsermons_speakers.gif";s:4:"1435";s:29:"icon_tx_wecsermons_topics.gif";s:4:"4d62";s:13:"locallang.php";s:4:"e737";s:16:"locallang_db.php";s:4:"e92e";s:47:"selicon_tx_wecsermons_sermons_record_type_0.gif";s:4:"02b6";s:47:"selicon_tx_wecsermons_sermons_record_type_1.gif";s:4:"02b6";s:47:"selicon_tx_wecsermons_sermons_record_type_2.gif";s:4:"02b6";s:47:"selicon_tx_wecsermons_sermons_record_type_3.gif";s:4:"02b6";s:7:"tca.php";s:4:"1016";s:16:".svn/all-wcprops";s:4:"dd93";s:12:".svn/entries";s:4:"04fb";s:11:".svn/format";s:4:"c30f";s:36:".svn/prop-base/ext_icon.gif.svn-base";s:4:"1131";s:61:".svn/prop-base/icon_tx_wecsermons_resource_types.gif.svn-base";s:4:"1131";s:56:".svn/prop-base/icon_tx_wecsermons_resources.gif.svn-base";s:4:"1131";s:54:".svn/prop-base/icon_tx_wecsermons_seasons.gif.svn-base";s:4:"1131";s:53:".svn/prop-base/icon_tx_wecsermons_series.gif.svn-base";s:4:"1131";s:54:".svn/prop-base/icon_tx_wecsermons_sermons.gif.svn-base";s:4:"1131";s:55:".svn/prop-base/icon_tx_wecsermons_speakers.gif.svn-base";s:4:"1131";s:53:".svn/prop-base/icon_tx_wecsermons_topics.gif.svn-base";s:4:"1131";s:71:".svn/prop-base/selicon_tx_wecsermons_sermons_record_type_0.gif.svn-base";s:4:"1131";s:71:".svn/prop-base/selicon_tx_wecsermons_sermons_record_type_1.gif.svn-base";s:4:"1131";s:71:".svn/prop-base/selicon_tx_wecsermons_sermons_record_type_2.gif.svn-base";s:4:"1131";s:71:".svn/prop-base/selicon_tx_wecsermons_sermons_record_type_3.gif.svn-base";s:4:"1131";s:33:".svn/text-base/ChangeLog.svn-base";s:4:"4842";s:34:".svn/text-base/README.txt.svn-base";s:4:"9386";s:63:".svn/text-base/class.tx_wecsermons_resourceTypeTca.php.svn-base";s:4:"4d64";s:55:".svn/text-base/class.tx_wecsermons_xmlView.php.svn-base";s:4:"3b4b";s:38:".svn/text-base/ext_emconf.php.svn-base";s:4:"13bb";s:36:".svn/text-base/ext_icon.gif.svn-base";s:4:"0e1f";s:41:".svn/text-base/ext_localconf.php.svn-base";s:4:"d9ae";s:38:".svn/text-base/ext_tables.php.svn-base";s:4:"d98a";s:38:".svn/text-base/ext_tables.sql.svn-base";s:4:"2c55";s:52:".svn/text-base/ext_typoscript_constants.txt.svn-base";s:4:"d41d";s:48:".svn/text-base/ext_typoscript_setup.txt.svn-base";s:4:"d41d";s:43:".svn/text-base/flexform_ds_pi1.xml.svn-base";s:4:"aa81";s:61:".svn/text-base/icon_tx_wecsermons_resource_types.gif.svn-base";s:4:"3a1a";s:56:".svn/text-base/icon_tx_wecsermons_resources.gif.svn-base";s:4:"86f1";s:54:".svn/text-base/icon_tx_wecsermons_seasons.gif.svn-base";s:4:"c430";s:53:".svn/text-base/icon_tx_wecsermons_series.gif.svn-base";s:4:"67c9";s:54:".svn/text-base/icon_tx_wecsermons_sermons.gif.svn-base";s:4:"d132";s:55:".svn/text-base/icon_tx_wecsermons_speakers.gif.svn-base";s:4:"1435";s:53:".svn/text-base/icon_tx_wecsermons_topics.gif.svn-base";s:4:"4d62";s:37:".svn/text-base/locallang.php.svn-base";s:4:"e737";s:40:".svn/text-base/locallang_db.php.svn-base";s:4:"e92e";s:71:".svn/text-base/selicon_tx_wecsermons_sermons_record_type_0.gif.svn-base";s:4:"02b6";s:71:".svn/text-base/selicon_tx_wecsermons_sermons_record_type_1.gif.svn-base";s:4:"02b6";s:71:".svn/text-base/selicon_tx_wecsermons_sermons_record_type_2.gif.svn-base";s:4:"02b6";s:71:".svn/text-base/selicon_tx_wecsermons_sermons_record_type_3.gif.svn-base";s:4:"02b6";s:31:".svn/text-base/tca.php.svn-base";s:4:"1016";s:23:".svn/tmp/tempfile.2.tmp";s:4:"4842";s:23:".svn/tmp/tempfile.3.tmp";s:4:"1016";s:23:".svn/tmp/tempfile.4.tmp";s:4:"123e";s:23:".svn/tmp/tempfile.5.tmp";s:4:"30eb";s:23:".svn/tmp/tempfile.6.tmp";s:4:"4842";s:23:".svn/tmp/tempfile.7.tmp";s:4:"1016";s:23:".svn/tmp/tempfile.8.tmp";s:4:"dfd9";s:36:"csh/locallang_csh_resource_types.xml";s:4:"2a24";s:31:"csh/locallang_csh_resources.xml";s:4:"7bcf";s:29:"csh/locallang_csh_seasons.xml";s:4:"ffa2";s:28:"csh/locallang_csh_series.xml";s:4:"f2fb";s:29:"csh/locallang_csh_sermons.xml";s:4:"8fae";s:30:"csh/locallang_csh_speakers.xml";s:4:"5452";s:28:"csh/locallang_csh_topics.xml";s:4:"ac36";s:16:"csh/.svn/entries";s:4:"4524";s:15:"csh/.svn/format";s:4:"c30f";s:60:"csh/.svn/text-base/locallang_csh_resource_types.xml.svn-base";s:4:"2a24";s:55:"csh/.svn/text-base/locallang_csh_resources.xml.svn-base";s:4:"7bcf";s:53:"csh/.svn/text-base/locallang_csh_seasons.xml.svn-base";s:4:"ffa2";s:52:"csh/.svn/text-base/locallang_csh_series.xml.svn-base";s:4:"f2fb";s:53:"csh/.svn/text-base/locallang_csh_sermons.xml.svn-base";s:4:"8fae";s:54:"csh/.svn/text-base/locallang_csh_speakers.xml.svn-base";s:4:"5452";s:52:"csh/.svn/text-base/locallang_csh_topics.xml.svn-base";s:4:"ac36";s:12:"doc/TODO.txt";s:4:"b550";s:13:"doc/Thumbs.db";s:4:"99f1";s:14:"doc/manual.sxw";s:4:"7beb";s:19:"doc/wizard_form.dat";s:4:"0173";s:20:"doc/wizard_form.html";s:4:"c4ce";s:20:"doc/.svn/all-wcprops";s:4:"7447";s:16:"doc/.svn/entries";s:4:"3a4f";s:15:"doc/.svn/format";s:4:"c30f";s:38:"doc/.svn/prop-base/manual.sxw.svn-base";s:4:"1131";s:36:"doc/.svn/text-base/TODO.txt.svn-base";s:4:"b550";s:38:"doc/.svn/text-base/manual.sxw.svn-base";s:4:"7beb";s:43:"doc/.svn/text-base/wizard_form.dat.svn-base";s:4:"0173";s:44:"doc/.svn/text-base/wizard_form.html.svn-base";s:4:"c4ce";s:13:"pi1/Thumbs.db";s:4:"d41b";s:14:"pi1/ce_wiz.gif";s:4:"085e";s:31:"pi1/class.tx_wecsermons_pi1.php";s:4:"7043";s:39:"pi1/class.tx_wecsermons_pi1_wizicon.php";s:4:"f89a";s:13:"pi1/clear.gif";s:4:"cc11";s:17:"pi1/locallang.php";s:4:"e270";s:19:"pi1/wecsermons.tmpl";s:4:"152d";s:20:"pi1/.svn/all-wcprops";s:4:"589b";s:16:"pi1/.svn/entries";s:4:"790b";s:15:"pi1/.svn/format";s:4:"c30f";s:38:"pi1/.svn/prop-base/ce_wiz.gif.svn-base";s:4:"1131";s:37:"pi1/.svn/prop-base/clear.gif.svn-base";s:4:"1131";s:38:"pi1/.svn/text-base/ce_wiz.gif.svn-base";s:4:"085e";s:55:"pi1/.svn/text-base/class.tx_wecsermons_pi1.php.svn-base";s:4:"6b26";s:63:"pi1/.svn/text-base/class.tx_wecsermons_pi1_wizicon.php.svn-base";s:4:"f89a";s:37:"pi1/.svn/text-base/clear.gif.svn-base";s:4:"cc11";s:41:"pi1/.svn/text-base/locallang.php.svn-base";s:4:"e270";s:43:"pi1/.svn/text-base/wecsermons.tmpl.svn-base";s:4:"152d";s:27:"pi1/.svn/tmp/tempfile.2.tmp";s:4:"b859";s:27:"pi1/.svn/tmp/tempfile.3.tmp";s:4:"c46a";s:28:"res/tt_news_v2_template.html";s:4:"e9b6";s:37:"res/tx_wec_sermons_resource_types.t3d";s:4:"1b65";s:28:"res/tx_wecsermons_styles.css";s:4:"0782";s:16:"res/.svn/entries";s:4:"0f0b";s:15:"res/.svn/format";s:4:"c30f";s:52:"res/.svn/text-base/tt_news_v2_template.html.svn-base";s:4:"e9b6";s:61:"res/.svn/text-base/tx_wec_sermons_resource_types.t3d.svn-base";s:4:"1b65";s:52:"res/.svn/text-base/tx_wecsermons_styles.css.svn-base";s:4:"0782";s:20:"static/constants.txt";s:4:"46a2";s:16:"static/setup.txt";s:4:"6c1b";s:23:"static/.svn/all-wcprops";s:4:"8574";s:19:"static/.svn/entries";s:4:"416f";s:18:"static/.svn/format";s:4:"c30f";s:44:"static/.svn/text-base/constants.txt.svn-base";s:4:"46a2";s:40:"static/.svn/text-base/setup.txt.svn-base";s:4:"6c1b";s:30:"static/.svn/tmp/tempfile.2.tmp";s:4:"6c1b";s:30:"static/.svn/tmp/tempfile.3.tmp";s:4:"6c1b";s:22:"static/style/setup.txt";s:4:"a1bc";s:25:"static/style/.svn/entries";s:4:"692b";s:24:"static/style/.svn/format";s:4:"c30f";s:46:"static/style/.svn/text-base/setup.txt.svn-base";s:4:"a1bc";}',
	'constraints' => array(
		'depends' => array(
			'php' => '4.3.0-0.0.0',
			'wec_api' => '0.9.1-0.0.0',
			'typo3' => '4.0.0-0.0.0',
		),
		'conflicts' => array(
		),
		'suggests' => array(
			'cron_rte_cleanenc' => '0.1.0-0.0.0',
		),
	),
	'suggests' => array(
	),
);

?>